### PR TITLE
feat(ui): logs search mvp 

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Contributes to NAN-908

- Implement v0 of the UI
It has listing and filtering on status. It's already huge (thanks react), so I don't want to commit to much and quickly this reviewed so I can add more stuff. I commented a few incomplete stuff.

http://localhost:3000/dev/logs

Screenshot 2024-05-10 at 16 14 41

### CODE REVIEW

File: packages/webapp/src/App.tsx [hunk 1]
```diff
@@ -53,25 +58,7 @@ const App = () => {
...
-                        wordBreak: 'break-all',
-                        whiteSpace: 'normal'
-                    },
-                    '.mantine-Prism-code': {
-                        fontFamily: 'Roboto Mono'
-                    }
-                })
-            }}
-        >
+        <MantineProvider theme={theme}>
```
Comment (bodinsamuel) [inline packages/webapp/src/App.tsx:61]: I initially started using Mantine for the new components so I upgraded it, but Radix was also already used and some stuff were missing/less convenient in Mantine

File: packages/webapp/src/pages/Logs/Show.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export const Show: React.FC = () => {
```
Comment (bodinsamuel) [inline packages/webapp/src/pages/Logs/Show.tsx:1]: Temp integration

File: packages/webapp/src/utils/utils.tsx [hunk 3]
```diff
@@ -146,6 +146,27 @@ export function formatDateToUSFormat(dateString: string): string {
     return formattedDate;
 }
 
+export function formatDateToIntFormat(dateString: string): string {
```
Comment (TBonnin) [inline packages/webapp/src/utils/utils.tsx:149]: nitpick: 
`formatDateToIntlFormat` maybe with a `L`. At first at read it as `To Integer Format`
or maybe something like
```
const Intl = {
    formatDate: (dateString: string): String { ...}
````

**Review by TBonnin — COMMENTED** [review]
I am probably underqualified to review React code. 😆  I'll let Khaliq approve it. Made a small comment

File: packages/webapp/src/utils/utils.tsx (hunk 3 above)
Comment (bodinsamuel) [reply packages/webapp/src/utils/utils.tsx:149]: ah yeah, I knew it was bad when writing it :D

**Comment by bodinsamuel** [comment]
> I am probably underqualified to review React code. 😆 I'll let Khaliq approve it. Made a small comment

Since he is a bit off in the next days, I'll appreciate an approval anyway ahah

**Review by TBonnin — APPROVED** [review]

**Comment by bodinsamuel** [comment]
Thanks (just to be clear I didn't meant to request a blind approval :D)

File: packages/webapp/src/pages/Logs/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            // We set this so it does not flicker when we go from a state of "filtered no records" to "default with records"...
+            // ...to not redisplay the empty state
+            setHasLogs(true);
+        }
+    }, [loading]);
+
+    if (error) {
+        return (
+            <DashboardLayout selectedItem={LeftNavBarItems.Logs} marginBottom={60}>
+                <Info color={error.error.code === 'feature_disabled' ? 'orange' : 'red'} classNames="text-xs" size={20}>
```
Comment (khaliqgant) [inline packages/webapp/src/pages/Logs/Search.tsx:42]: This text is super small -- haven't checked the design but based on the app relative to everything else it seems too small IMO.

File: packages/webapp/src/pages/Logs/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    if (data.pagination.total <= 0 && !hasLogs) {
+        return (
+            <DashboardLayout selectedItem={LeftNavBarItems.Logs} marginBottom={60}>
+                <h2 className="text-3xl font-semibold text-white mb-4">Logs</h2>
+
+                <div className="flex flex-col border border-zinc-500 rounded items-center text-white text-center py-24 gap-2">
+                    <h2 className="text-xl">You don&apos;t have logs yet.</h2>
+                    <div className="text-sm text-zinc-400">Note that logs older than 15days are automatically cleared.</div>
```
Comment (khaliqgant) [inline packages/webapp/src/pages/Logs/Search.tsx:66]: add space between 15 and days

File: packages/webapp/src/pages/Logs/components/OperationRow.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import { flexRender } from '@tanstack/react-table';
+import type { SearchLogsData } from '@nangohq/types';
+
+import { Drawer, DrawerContent, DrawerTrigger } from '../../../components/ui/Drawer';
+import * as Table from '../../../components/ui/Table';
+import { Show } from '../Show';
+
+export const OperationRow: React.FC<{ row: Row<SearchLogsData> }> = ({ row }) => {
+    return (
+        <Drawer direction="right" snapPoints={['1034px']} handleOnly={true} noBodyStyles={true}>
```
Comment (khaliqgant) [inline packages/webapp/src/pages/Logs/components/OperationRow.tsx:11]: 1034 is a new width? Should this be a const? Again, haven't checked designs so not sure

File: packages/webapp/src/pages/Logs/components/StatusTag.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import type { SearchLogsData } from '@nangohq/types';
+
+export const StatusTag: React.FC<{ state: SearchLogsData['state'] }> = ({ state }) => {
+    if (state === 'success') {
+        return (
+            <div className="inline-flex px-1 pt-[1px] bg-state-green-900 bg-opacity-30 rounded">
```
Comment (khaliqgant) [inline packages/webapp/src/pages/Logs/components/StatusTag.tsx:6]: why is this `pt` when nothing else is?

File: packages/webapp/tailwind.config.js [hunk 8]
```diff
@@ -30,6 +31,9 @@ module.exports = {
                 'dark-700': '#18181B',
                 'dark-800': '#09090B',
                 'bg-dark-blue': '#182633',
+                'state-green-900': '#84D65A',
```
Comment (khaliqgant) [inline packages/webapp/tailwind.config.js:34]: what does state green mean and why is 400 the same as 900?

File: packages/webapp/tailwind.config.js [hunk 9]
```diff
@@ -30,6 +31,9 @@ module.exports = {
...
+                'state-green-400': '#84D65A',
+                'row-hover': '#0d0d14',
```
Comment (khaliqgant) [inline packages/webapp/tailwind.config.js:36]: `row-hover` seems very generic

File: packages/webapp/src/pages/Logs/Search.tsx (hunk 4 above)
Comment (bodinsamuel) [reply packages/webapp/src/pages/Logs/Search.tsx:42]: yes agreed, I have already mentioned that to Gabrielle

File: packages/webapp/tailwind.config.js (hunk 8 above)
Comment (bodinsamuel) [reply packages/webapp/tailwind.config.js:34]: lack of better word, it's used for the state (status) of logs.
good catch on 400 thx

File: packages/webapp/tailwind.config.js (hunk 9 above)
Comment (bodinsamuel) [reply packages/webapp/tailwind.config.js:36]: yep, I lacked imagination happy to change if you have something. It's for the hover of table's row

File: packages/webapp/src/pages/Logs/components/OperationRow.tsx (hunk 6 above)
Comment (bodinsamuel) [reply packages/webapp/src/pages/Logs/components/OperationRow.tsx:11]: it's the only to get a fixed drawer unfortunately, probably should be a const yes

File: packages/webapp/src/pages/Logs/components/StatusTag.tsx (hunk 7 above)
Comment (bodinsamuel) [reply packages/webapp/src/pages/Logs/components/StatusTag.tsx:6]: indeed thanks

**Pull request merged**